### PR TITLE
feat(crane): add prefix routing & stripping

### PR DESCRIPTION
**Pull request by nandor-magyar**
Two things here:
- added path routing implementation to crane
- fix for crane's domain generation

### CODE REVIEW

File: golang/internal/domain/domain.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package domain
+
+import (
+	"github.com/dyrector-io/dyrectorio/golang/internal/config"
+	"github.com/dyrector-io/dyrectorio/golang/internal/util"
+)
+
+func GetHostRule(hostPrefix, hostRoot, containerName, prefix string, cfg *config.CommonConfiguration) string {
```
Comment (polaroi8d) [inline golang/internal/domain/domain.go:8]: Where is the unit tests? 🦅

File: golang/internal/mapper/grpc.go [hunk 2]
```diff
@@ -284,7 +284,11 @@ func mapVolumes(in []*agent.Volume) []v1.Volume {
 		}
 
 		if in[i].Type != nil {
-			volume.Type = in[i].Type.String()
+			if *in[i].Type == common.VolumeType_MEM || *in[i].Type == common.VolumeType_TMP {
+				volume.Type = strings.ToLower(in[i].Type.String())
```
Comment (polaroi8d) [inline golang/internal/mapper/grpc.go:288]: Why do we need this?

File: golang/internal/domain/domain.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	domain := []string{}
+	rootDomain := ""
+
+	if hostRoot != "" {
+		rootDomain = hostRoot
+	} else if cfg.RootDomain != "" {
+		rootDomain = cfg.RootDomain
+	}
+
+	// generate the name.prefix.domain from prefix and container name
```
Comment (polaroi8d) [inline golang/internal/domain/domain.go:18]: What is "name", be more specific please.

File: golang/internal/mapper/grpc.go (hunk 2 above)
Comment (nandor-magyar) [reply golang/internal/mapper/grpc.go:288]: This was incorrect, **RWO**, **RWX** and **RW** are valid "real" storage types, while **tmp** and **memory** are synthetic ones. Around agents these are the values, unfortunately proto enum is always capitalized. This breaks no compatibility and works still.

File: golang/internal/domain/domain.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/dyrector-io/dyrectorio/golang/internal/util"
+)
+
+func GetHostRule(hostPrefix, hostRoot, containerName, prefix string, cfg *config.CommonConfiguration) string {
+	domain := []string{}
+	rootDomain := ""
+
+	if hostRoot != "" {
+		rootDomain = hostRoot
+	} else if cfg.RootDomain != "" {
```
[the remark this answers is not in the document]
Comment (nandor-magyar) [reply golang/internal/domain/domain.go:14]: Yes

File: golang/internal/domain/domain.go (hunk 3 above)
Comment (nandor-magyar) [reply golang/internal/domain/domain.go]: Added more comments around there.

File: golang/internal/domain/domain.go (hunk 1 above)
Comment (nandor-magyar) [reply golang/internal/domain/domain.go]: Added

**Review by polaroi8d — APPROVED** [review]
LGTM, good job!

**Pull request merged**